src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/solicitorcreatecase/mandatorydatavalidation/ApplicantSolicitorDetailsValidator.java

```diff
@@ -0,0 +1,47 @@
+package uk.gov.hmcts.reform.finrem.caseorchestration.handler.solicitorcreatecase.mandatorydatavalidation;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.FinremCaseData;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.YesOrNo;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.wrapper.ContactDetailsWrapper;
+import uk.gov.hmcts.reform.finrem.caseorchestration.utils.NullChecker;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+@Slf4j
+class ApplicantSolicitorDetailsValidator implements MandatoryDataValidator {
+    @Override
+    public List<String> validate(FinremCaseData caseData) {
+        ContactDetailsWrapper contactDetailsWrapper = caseData.getContactDetailsWrapper();
+
+        List<String> ret = new ArrayList<>();
+
+        if (caseData.isConsentedApplication()) {
+            if (YesOrNo.NO.equals(contactDetailsWrapper.getApplicantRepresented())) {
+                log.info("{} - Skip validating solicitor details since the applicant is not represented", caseData.getCcdCaseId());
+                return ret;
+            }
+            if (contactDetailsWrapper.getSolicitorAddress() == null
+                || !NullChecker.anyNonNull(contactDetailsWrapper.getSolicitorAddress())) {
+                ret.add("Applicant solicitor's address is required.");
+            }
+            if (!StringUtils.hasText(contactDetailsWrapper.getSolicitorEmail())) {
+                ret.add("Applicant solicitor's email is required.");
+            }
+            if (!StringUtils.hasText(contactDetailsWrapper.getSolicitorPhone())) {
+                ret.add("Applicant solicitor's phone is required.");
+            }
+            if (!StringUtils.hasText(contactDetailsWrapper.getSolicitorFirm())) {
+                ret.add("Applicant solicitor's name of your firm is required.");
+            }
+            if (!StringUtils.hasText(contactDetailsWrapper.getSolicitorName())) {
```

Comment:
We can potentially make a method out of this. Something like this?

```
    private void validateField(String fieldValue, String fieldName, List<String> errors) {
        if (!StringUtils.hasText(fieldValue)) {
            errors.add(String.format("Applicant solicitor's %s is required.", fieldName));
        }
    }
```

Reply:
@hamzahtahirhmcts Good idea. Fixed by https://github.com/hmcts/finrem-case-orchestration-service/pull/1800#:~:text=the%20common%20logic-,909aac8,-Merge%20state